### PR TITLE
Make "neon_local stop" less verbose.

### PR DESCRIPTION
**Pull request by hlinnaka**
I got annoyed by all the noise in CI test output.

Before:

    $ ./target/release/neon_local stop
    Stop pageserver gracefully
    Pageserver still receives connections
    Pageserver stopped receiving connections
    Pageserver status is: Reqwest error: error sending request for url (http://127.0.0.1:9898/v1/status): error trying to connect: tcp connect error: Connection refused (os error 111)
    initializing for sk 1 for 7676
    Stop safekeeper gracefully
    Safekeeper still receives connections
    Safekeeper stopped receiving connections
    Safekeeper status is: Reqwest error: error sending request for url (http://127.0.0.1:7676/v1/status): error trying to connect: tcp connect error: Connection refused (os error 111)

After:

    $ ./target/release/neon_local stop
    Stopping pageserver gracefully...done!
    Stopping safekeeper 1 gracefully...done!

Also removes the spurious "initializing for sk 1 for 7676" message from
"neon_local start"

### CODE REVIEW

**Review by aome510 — APPROVED** [review]
LGTM! One minor concern: it's not clear to me why do we need `io::stdout().flush().unwrap();`?

File: control_plane/src/safekeeper.rs [hunk 1]
```diff
@@ -250,25 +249,35 @@ impl SafekeeperNode {
...
         for _ in 0..100 {
-            if let Err(_e) = TcpStream::connect(&address) {
-                println!("Safekeeper stopped receiving connections");
-
-                //Now check status
+            if !tcp_stopped {
+                if let Err(err) = TcpStream::connect(&address) {
+                    tcp_stopped = true;
+                    if err.kind() != io::ErrorKind::ConnectionRefused {
+                        println!("\nSafekeeper connection failed with error: {err}");
```
Comment (aome510) [inline control_plane/src/safekeeper.rs:258]: If this is unexpected error, we should use `eprintln!` instead.

File: control_plane/src/safekeeper.rs [hunk 2]
```diff
@@ -250,25 +249,35 @@ impl SafekeeperNode {
...
-                    Ok(_) => {
-                        println!("Safekeeper status is OK. Wait a bit.");
-                        thread::sleep(Duration::from_secs(1));
+                    Err(SafekeeperHttpError::Transport(err)) if err.is_connect() => {
+                        println!("done!");
+                        return Ok(());
                     }
                     Err(err) => {
-                        println!("Safekeeper status is: {}", err);
+                        println!("\nSafekeeper status check failed with error: {err}");
```
Comment (aome510) [inline control_plane/src/safekeeper.rs:270]: Same here.

File: control_plane/src/storage.rs [hunk 3]
```diff
@@ -287,25 +288,36 @@ impl PageServerNode {
...
         // if pageserver checkpoints a lot of data
+        let mut tcp_stopped = false;
         for _ in 0..100 {
-            if let Err(_e) = TcpStream::connect(&address) {
-                println!("Pageserver stopped receiving connections");
+            if !tcp_stopped {
+                if let Err(err) = TcpStream::connect(&address) {
+                    tcp_stopped = true;
+                    if err.kind() != io::ErrorKind::ConnectionRefused {
+                        println!("\nPageserver connection failed with error: {err}");
```
Comment (aome510) [inline control_plane/src/storage.rs:297]: Same here.

File: control_plane/src/storage.rs [hunk 4]
```diff
@@ -287,25 +288,36 @@ impl PageServerNode {
...
-                    Ok(_) => {
-                        println!("Pageserver status is OK. Wait a bit.");
-                        thread::sleep(Duration::from_secs(1));
+                    Err(PageserverHttpError::Transport(err)) if err.is_connect() => {
+                        println!("done!");
+                        return Ok(());
                     }
                     Err(err) => {
-                        println!("Pageserver status is: {}", err);
+                        println!("\nPageserver status check failed with error: {err}");
```
Comment (aome510) [inline control_plane/src/storage.rs:310]: Same here.

**Comment by hlinnaka** [comment]
> LGTM! One minor concern: it's not clear to me why do we need `io::stdout().flush().unwrap();`?

Stdout is usually line-buffered. The loop prints a dot once per second, with `print(".")`, as a simple progress indicator. Without the flush, the dots would not be printed to the terminal until the next newline.

**Pull request merged**